pulp_rpm/app/migrations/0046_modulemd_fields.py

```diff
@@ -0,0 +1,60 @@
+# Generated by Django 3.2.15 on 2022-08-29 16:33
+
+from django.db import migrations, models, transaction
+
+import gi
+
+gi.require_version("Modulemd", "2.0")
+from gi.repository import Modulemd as mmdlib  # noqa: E402
+
+
+def populate_new_fields(apps, schema_editor):
+    """Create snippet from artifact and remove artifact."""
+    Modulemd = apps.get_model('rpm', 'Modulemd')
+    modules_to_update = []
+
+    for modulemd in Modulemd.objects.filter(profiles={}):
+        module_index = mmdlib.ModuleIndex.new()
+        try:
+            ret, err = module_index.update_from_string(modulemd.snippet, False)
+            if err:
+                raise ValueError(
+                    "Repair failed on module "
+                    f"'{modulemd.name}.{modulemd.stream}.{modulemd.version}'."
+                )
+            for module in module_index.get_module_names():
+                for stream in module_index.get_module(module).get_all_streams():
+                    modulemd.profiles = {}
+                    modulemd.description = stream.get_description()
+                    for profile in stream.get_profile_names():
+                        stream_profile = stream.get_profile(profile)
+                        modulemd.profiles[profile] = stream_profile.get_rpms()
+                        modules_to_update.append(modulemd)
+        except ValueError:
+            # Due to issue #2735 it could happen that snippet will be empty
+            # https://github.com/pulp/pulp_rpm/issues/2735
```

Comment:
We do need to fix this if it's true BTW.  I don't think we should ignore it silently.  Let's print a warning at least.